src/featureconverter.js

```diff
@@ -397,38 +397,69 @@ olcs.FeatureConverter.prototype.olPolygonGeometryToCesium =
   olGeometry = olcs.core.olGeometryCloneTo4326(olGeometry, projection);
   goog.asserts.assert(olGeometry.getType() == 'Polygon');
 
-  var rings = olGeometry.getLinearRings();
-  // always update Cesium externs before adding a property
-  var hierarchy = {};
-  var polygonHierarchy = hierarchy;
-  goog.asserts.assert(rings.length > 0);
-
-  for (var i = 0; i < rings.length; ++i) {
-    var olPos = rings[i].getCoordinates();
-    var positions = olcs.core.ol4326CoordinateArrayToCsCartesians(olPos);
-    goog.asserts.assert(positions && positions.length > 0);
-    if (i == 0) {
-      hierarchy.positions = positions;
-    } else {
-      hierarchy.holes = {
-        // always update Cesium externs before adding a property
-        positions: positions
-      };
-      hierarchy = hierarchy.holes;
+  var fillGeometry, outlineGeometry;
+  if (feature.getGeometry().get('olcs.polygon_kind') === 'rectangle') {
```

Comment:
You should only enter here if there are exactly 4 elements in the coordinates array.